js_modules/dagster-ui/packages/ui-core/src/runs/Run.tsx

```diff
@@ -99,23 +101,31 @@ export const Run = (props: RunProps) => {
   return (
     <RunContext.Provider value={run}>
       <LogsProvider key={runId} runId={runId}>
-        {(logs) => (
-          <RunMetadataProvider logs={logs}>
-            {(metadata) => (
-              <RunWithData
-                run={run}
-                runId={runId}
-                logs={logs}
-                logsFilter={logsFilter}
-                metadata={metadata}
-                selectionQuery={selectionQuery}
-                onSetLogsFilter={setLogsFilter}
-                onSetSelectionQuery={onSetSelectionQuery}
-                onShowStateDetails={onShowStateDetails}
-              />
-            )}
-          </RunMetadataProvider>
-        )}
+        {(logs) => {
+          function Wrapper() {
+            React.useLayoutEffect(() => {
+              trace.onLogsLoaded();
+            }, []);
+            return (
```

Comment:
!! same call-out as the other PR, but anytime Run.tsx state changes it will destroy this entire subtree and rebuild it from scratch because of the inline component definition. I think we should change the way this works before shipping this.

Reply:
ugh, this is why I don't like the child render function pattern

Reply:
But also because this was already using a render prop pattern, moving this Wrapper outside of the component actually won't accomplish anything.